dcpy/connectors/socrata/metadata.py

```diff
@@ -0,0 +1,129 @@
+from pathlib import Path
+import typer
+import yaml
+
+from dcpy.connectors.socrata import publish as pub
+from dcpy.connectors.socrata import metadata
+
+from dcpy.metadata import models
+from dcpy.utils.logging import logger
+
+soc_types_to_dcp_types = {
+    "text": "text",
+    "multipolygon": "wkb",
+    "calendar_date": "datetime",
+}
+
+
+def make_dcp_col(c: pub.Socrata.Responses.Column):
+    samples = c.get("cachedContents", {}).get("top", [])
+    sample = "<FILL_ME_IN>"
+    dcp_type = None
+
+    if samples:
+        sample = samples[0].get("item")
+
+    if c["renderTypeName"] == "number":
+        if type(sample) == float:
+            dcp_type = "double"
+        elif type(sample) == int:
+            dcp_type = "integer"
+        else:
+            dcp_type = "<FILL_ME_IN>!"
+    else:
+        dcp_type = soc_types_to_dcp_types[c["renderTypeName"]]
+
+    dcp_col = {
+        "name": c["fieldName"],
+        "display_name": c["name"],
+        "description": c["description"],
+        "data_type": dcp_type,
+        "example": sample,
+    }
+    if c["renderTypeName"] == "multipolygon":
+        dcp_col["readme_data_type"] = "geometry"
+
+    ENTIRELY_ARBITRARY_MAX_STANDARDIZED_VALS = 10
+    if (
+        samples
+        and len(samples) < ENTIRELY_ARBITRARY_MAX_STANDARDIZED_VALS
+        and len(samples) == int(c["cachedContents"].get("cardinality"))  # type: ignore
```

Comment:
Not that everything needs to be 100% type safe, in particular when working with json/dict responses from apis, but just curious on your thoughts between pydantic types and typeddict for these objects, having done a bit of work with them

Reply:
That's a good question... In this case, I was using typeddict more for documentation purposes than anything, but it might not be a bad idea to just default to one or the other.